### PR TITLE
container-runtimes: include note about latest validated docker

### PR DESCRIPTION
**Pull request by enyinna1234**
This commit fixes the packages not found error during
Docker installation. The packages containerd.io 1.2.13-2,
docker-ce 19.03.11, and docker-cs-cli 19.03.11 are not in
Ubuntu 20.10. This commit instead installs containerd.io 1.4.3-1,
docker-ce 20.10.5, and docker-ce-cli 20.10.5.

fixes https://github.com/kubernetes/kubernetes/issues/99831

Signed-off-by: Enyinna Ochulor <eochulor@vmware.com>

### CODE REVIEW

**Comment by enyinna1234** [comment]
Closes https://github.com/kubernetes/kubernetes/issues/99831

**Comment by neolit123** [comment]
@bart0sh @enyinna1234 

IIRC, we used to pin this version because particular versions of docker had issues, which i don't think is the case anymore.
(distro versions are patched).

there are some open related PRs:
https://github.com/kubernetes/website/pull/26952
https://github.com/kubernetes/website/issues/26906

in the above PRs we are recommending to link to the docker docs per distro and we are removing the per-distro-version tabs.

i think we should just mention that the latest validated version of docker against k8s is X.
in this case it's 20.10:
https://github.com/kubernetes/kubernetes/pull/98977

let's hold this change until the above two PRs merge and after that we can amend this latest validated version.

**Comment by neolit123** [comment]
@enyinna1234 

the page changed after the other PR merged:
https://github.com/kubernetes/website/blob/master/content/en/docs/setup/production-environment/container-runtimes.md#docker

i suggest that instead of pinning a version, we should mentioned what is the latest validated version.

proposing that this line:
> On each of your nodes, install the Docker for your Linux distribution as per Install Docker Engine

becomes:
> On each of your nodes, install the Docker for your Linux distribution as per Install Docker Engine. You can find the latest validated version of Docker in this [dependencies.yaml file](https://git.k8s.io/kubernetes/build/dependencies.yaml).


please also add
```
fixes https://github.com/kubernetes/kubernetes/issues/99831
```

in this PRs description.

**Review by neolit123 — COMMENTED** [review]
thanks
/lgtm
/sig cluster-lifecycle

**Comment by neolit123** [comment]
@enyinna1234 

> please also add
> 
> ```
> fixes https://github.com/kubernetes/kubernetes/issues/99831
> ```
> 
> in this PRs description.

**Comment by neolit123** [comment]
/retitle container-runtimes: include note about latest validated docker

**Review by pacoxu — APPROVED** [review]
/lgtm
/assign @tengqm
for approval

**Comment by annajung** [comment]
thanks!
/approve

**Pull request merged**
